src/Service/AbstractPostSearchService.php

```diff
@@ -70,44 +70,40 @@ protected function giveSetResults($query, array $opts = [])
     protected function getQuerySql($searchText, $postTypes = [], $action = "SELECT")
     {
         global $wpdb;
+        $queryStr = NULL;
 
-        if ($action == 'COUNT') {
-            $queryStr = "SELECT COUNT(*) as cpt";
-        } else {
-            $queryStr = "SELECT $wpdb->posts.*, MATCH (" . implode(',', $this->getIndexedFields()) . ") AGAINST ('" . $searchText . "' IN NATURAL LANGUAGE MODE) as score";
-        }
-        $queryStr .= "
+        if ($searchText) {
+            // Search for longer sentences
+            $searchText = '*' . $searchText . '*';
+
+            if ($action == 'COUNT') {
+                $queryStr = "SELECT COUNT(*) as cpt";
+            } else {
+                $queryStr = "SELECT $wpdb->posts.*, MATCH (" . implode(',', $this->getIndexedFields()) . ") AGAINST ('" . $searchText . "' IN BOOLEAN MODE) as score";
+            }
+            $queryStr .= "
                     FROM $wpdb->posts";
-        $queryStr .= "
-                    WHERE 1 ";
-        $queryStr .= " AND (";
-        $queryStr .= "$wpdb->posts.post_status = 'publish' OR $wpdb->posts.post_status = 'private'";
-        if (in_array('attachment', $postTypes)) {
-            $queryStr .= " OR $wpdb->posts.post_status = 'inherit'";
-        }
-        $queryStr .= ")";
-        if (!empty($postTypes)) {
             $queryStr .= "
+                    WHERE 1 ";
+            $queryStr .= " AND (";
+            $queryStr .= "$wpdb->posts.post_status = 'publish' OR $wpdb->posts.post_status = 'private'";
```

Comment:
Pendant qu'on est dans ce fichier, il faudrait refacto cette ligne pour pouvoir passer un tableau de status à chercher en attribut statique de la classe peut-être puis mettre à dispo un filter sur ce tableau. et transformer la requête pour utiliser IN plutôt. 

Du coup on obtiendrait qqh du genre :
$wpdb->posts.post_status IN apply_filters('wwp-search.post_search.searchable.post.status',self::$searchablePostStatus,$this)